gammapy/maps/axes.py

```diff
@@ -0,0 +1,368 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import copy
+import inspect
+import numpy as np
+from astropy.time import Time
+import astropy.units as u
+from gammapy.utils.interpolation import interpolation_scale
+from .utils import INVALID_INDEX
+
+__all__ = ["TimeMapAxis"]
+
+
+class TimeMapAxis:
+    """Class representing a time axis.
+
+    Provides methods for transforming to/from axis and pixel coordinates.
+    A time axis can represent non-contiguous sequences of non-overlapping time intervals.
+
+    Time intervals must be provided in increasing order.
+
+    Parameters
+    ----------
+    edges_min : `~astropy.units.Quantity`
+        Array of edge time values. This the time delta w.r.t. to the reference time.
+    edges_max : ``~astropy.units.Quantity`
+        Array of edge time values. This the time delta w.r.t. to the reference time.
+    reference_time : `~astropy.time.Time`
+        Reference time to use.
+    name : str
+        Axis name
+    interp : str
+        Interpolation method used to transform between axis and pixel
+        coordinates.  For now only 'lin' is supported.
+    """
+    node_type = "intervals"
+    def __init__(self, edges_min, edges_max, reference_time, name="time", interp="lin"):
+        self._name = name
+
+        edges_min = u.Quantity(edges_min, ndmin=1)
+        edges_max = u.Quantity(edges_max, ndmin=1)
+
+        if not isinstance(reference_time, Time):
```

Comment:
Just call `Time(reference_time)` here and let Astropy raise the error...

Reply:
Done